### PR TITLE
Expire at

### PR DESCRIPTION
**Pull request by AlexeyVavilin**
- ABI v2 and header params supported
- Message expiration waiting

### CODE REVIEW

File: src/modules/TONContractsModule.js [hunk 1]
```diff
@@ -470,17 +504,63 @@ export default class TONContractsModule extends TONModule implements TONContract
...
         parentSpan?: (Span | SpanContext),
     ): Promise<QTransaction> {
+
         const messageId = await this.sendMessage(message, parentSpan);
-        const transaction: QTransaction = await this.queries.transactions.waitFor({
-            in_msg: { eq: messageId },
-            status: { eq: QTransactionProcessingStatus.finalized },
-        }, resultFields, 40_000, parentSpan);
+
+        let timeout = DEFAULT_TIMEOUT;
```
Comment (melsomino) [inline src/modules/TONContractsModule.js]: why you don't use timeout from config?

File: src/modules/TONQueriesModule.js [hunk 2]
```diff
@@ -400,7 +402,7 @@ class TONQueriesModuleCollection implements TONQCollection {
         const params = findParams<TONWaitForParams>(args, 'filter');
         const filter = params ? params.filter : args[0];
         const result: string = params ? params.result : (args[1]: any);
-        const timeout = (params ? params.timeout : args[2]) || 40000;
+        const timeout = (params ? params.timeout : args[2]) || DEFAULT_TIMEOUT;
```
Comment (melsomino) [inline src/modules/TONQueriesModule.js]: may be better to retrieve this from config?

File: types.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ export type TONConfigData = {
     queriesWsServer?: string,
     log_verbose?: boolean,
     tracer?: ?Object, // opentracing.Tracer
+    authorization?: string,
```
Comment (melsomino) [inline types.js]: authorization??
it is superseded by `accessKey`
and remove please obsolete `*server` parameters

Comment (AlexeyVavilin) [reply types.js]: Done

File: src/modules/TONContractsModule.js [hunk 4]
```diff
@@ -470,17 +504,63 @@ export default class TONContractsModule extends TONModule implements TONContract
...
+            timeout = expire * 1000 - Date.now() + DEFAULT_TIMEOUT;
+            
+            if (timeout <= 0) {
+                throw TONClientError.sendNodeRequestFailed("Message already expired");
+            }
+
+            const waitExpired = async () => {
+                // wait for block, produced after `expire` to guarantee that message is rejected
+                const block: QBlock = await this.queries.blocks.waitFor({
+                        workchain_id: { eq: -1 },
```
[the remark this answers is not in the document]
Comment (AlexeyVavilin) [reply src/modules/TONContractsModule.js]: Done

**Pull request merged**